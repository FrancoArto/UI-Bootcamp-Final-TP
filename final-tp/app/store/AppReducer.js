import { combineReducers } from 'redux';
import tweetsReducer from './tweets/tweetsReducer'
import trendsReducer from './trends/trendsReducer'
import settingsReducer from './settings/settingsReducer'
import userReducer  from './users/userReducer'

const AppReducer = combineReducers({
  tweetsReducer,
  trendsReducer,
  settingsReducer,
  userReducer  
});

export default AppReducer;