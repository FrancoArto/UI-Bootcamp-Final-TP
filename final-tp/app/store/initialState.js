const tweetsInitialState = {
    data: [],
    loading: false,
    loadingSingleTweet: false,
    error: null,
    searchText: null,
    searchResults: [],
    userTimeline: [],
    singleTweetId: null,
    singleTweet: {}
};

const userInitialState = {
    loading: false,
    error: null,
    userId: null,
    userData: {}
}

const trendsInitialState = {
    data: [],
    loading: false,
    error: null,
}

//False is for 'not checked'
const appViewsConfigs = {
    verified: false,
    following: false,
    defaultInfo: false,
    withLink: false,
    withTruncatedText: false
}


export { tweetsInitialState, appViewsConfigs, userInitialState, trendsInitialState };