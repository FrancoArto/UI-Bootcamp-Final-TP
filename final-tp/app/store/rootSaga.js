import { all } from 'redux-saga/effects'
import {trendsSaga} from './trends/trendsSaga'
import { userSaga } from './users/userSaga';
import { tweetsSaga } from './tweets/tweetsSaga';

export default function* rootSaga() {
  yield all([
    tweetsSaga(),
    trendsSaga(),
    userSaga()
  ])
}
