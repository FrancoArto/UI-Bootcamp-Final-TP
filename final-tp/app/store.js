import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga' 
import AppReducer from './reducers/index'; //Import the reducer

const sagaMiddleware = createSagaMiddleware()
 
export default createStore(AppReducer, applyMiddleware(sagaMiddleware));
