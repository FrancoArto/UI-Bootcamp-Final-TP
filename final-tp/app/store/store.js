import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import AppReducer from '../reducers/index'; //Import the reducer
import rootSaga from '../sagas/index';
import reactotron from '../reactotronConfig'

const sagaMonitor = reactotron.createSagaMonitor()


const sagaMiddleware = createSagaMiddleware({ sagaMonitor })

export default reactotron.createStore(AppReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga)
