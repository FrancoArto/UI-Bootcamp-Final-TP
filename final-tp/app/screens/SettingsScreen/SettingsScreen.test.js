import React from 'react';
import {mount, shallow} from 'enzyme'
import store from '../../store/store';
import ReduxSettingsScreen, {SettingsScreen} from './SettingsScreen';



it('renders without crashing', () => {
  const rendered = shallow(<SettingsScreen />)
  expect(rendered).toMatchSnapshot();
});

it('renders connected to store', () => {
  const rendered = shallow(<ReduxSettingsScreen store={store} />)
  expect(rendered).toMatchSnapshot();
});

it('handles verified setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  } 

  const settingsChanged = jest.fn()

  const wrapper = mount(<SettingsScreen
    settings={inputSettings}
    SettingsChanged={settingsChanged}/>)

  const setting = wrapper.find('Settings').first();
  setting.props().onVerifiedChange()
  expect(settingsChanged).toHaveBeenCalled()
});

it('handles following setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }  

  const settingsChanged = jest.fn()

  const wrapper = mount(<SettingsScreen
    settings={inputSettings}
    SettingsChanged={settingsChanged}/>)

  const setting = wrapper.find('Settings').first();
  setting.props().onFollowingChange()
  expect(settingsChanged).toHaveBeenCalled()
});

it('handles default info setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }  

  const settingsChanged = jest.fn()

  const wrapper = mount(<SettingsScreen
    settings={inputSettings}
    SettingsChanged={settingsChanged}/>)

  const setting = wrapper.find('Settings').first();
  setting.props().onDefaultInfoChange()
  expect(settingsChanged).toHaveBeenCalled()
});

it('handles with link setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }  

  const settingsChanged = jest.fn()

  const wrapper = mount(<SettingsScreen
    settings={inputSettings}
    SettingsChanged={settingsChanged}/>)

  const setting = wrapper.find('Settings').first();
  setting.props().onWithLinkChange()
  expect(settingsChanged).toHaveBeenCalled()
});

it('handles truncated text setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }  

  const settingsChanged = jest.fn()

  const wrapper = mount(<SettingsScreen
    settings={inputSettings}
    SettingsChanged={settingsChanged}/>)

  const setting = wrapper.find('Settings').first();
  setting.props().onWithTruncatedTextChange()
  expect(settingsChanged).toHaveBeenCalled()
});