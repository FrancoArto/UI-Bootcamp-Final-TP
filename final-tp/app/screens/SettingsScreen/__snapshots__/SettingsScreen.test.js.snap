// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders connected to store 1`] = `
<SettingsScreen
  SettingsChanged={[Function]}
  settings={
    Object {
      "defaultInfo": false,
      "following": false,
      "verified": false,
      "withLink": false,
      "withTruncatedText": false,
    }
  }
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`renders without crashing 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Settings
    onDefaultInfoChange={[Function]}
    onFollowingChange={[Function]}
    onVerifiedChange={[Function]}
    onWithLinkChange={[Function]}
    onWithTruncatedTextChange={[Function]}
  />
</Component>
`;
